Buildings/Experimental/OpenBuildingControl/ASHRAE/G36/Atomic/EconEnableDisableMultiZone.mo

```diff
@@ -4,83 +4,84 @@ block EconEnableDisableMultiZone
     annotation (Placement(transformation(extent={{-220,30},{-180,70}}),
         iconTransformation(extent={{-120,10},{-100,30}})));
   CDL.Interfaces.IntegerInput uZonSta "Zone state signal"
     annotation (Placement(transformation(extent={{-220,-30},{-180,10}}),
         iconTransformation(extent={{-120,-10},{-100,10}})));
 
-  CDL.Interfaces.RealOutput yOutDamPosMax "Maximum outdoor air damper position"
+  CDL.Interfaces.RealOutput yOutDamPosMax(final min=0, final max=1) "Maximum outdoor air damper position"
     annotation (Placement(transformation(extent={{180,-150},{200,-130}}),
         iconTransformation(extent={{100,28},{140,68}})));
-  CDL.Interfaces.RealOutput yRetDamPosMin "Minimum return air damper position"
+  CDL.Interfaces.RealOutput yRetDamPosMin(final min=0, final max=1) "Minimum return air damper position"
     annotation (Placement(transformation(extent={{180,-260},{200,-240}}),
         iconTransformation(extent={{100,-100},{140,-60}})));
-  CDL.Interfaces.RealOutput yRetDamPosMax "Maximum return air damper position"
+  CDL.Interfaces.RealOutput yRetDamPosMax(final min=0, final max=1) "Maximum return air damper position"
     annotation (Placement(transformation(
         extent={{180,-220},{200,-200}}), iconTransformation(extent={{100,-40},{140,0}})));
 
-  CDL.Logical.TrueFalseHold TrueFalseHold(duration=600) "10 min on/off delay"
+  CDL.Logical.TrueFalseHold trueFalseHold(final duration=600) "10 min on/off delay"
     annotation (Placement(transformation(extent={{0,200},{20,220}})));
-  CDL.Logical.And3 andEnaDis "Logical \"and\" checks freeze protection stage and zone state"
+  CDL.Logical.And3 andEnaDis "Logical and to check freeze protection stage and zone state"
    annotation (Placement(transformation(extent={{40,30},{60,50}})));
 
 protected
-  parameter Real uTemHigLimCutHig(final unit="K", quantity = "ThermodynamicTemperature") = 0
+  final parameter Modelica.SIunits.Temperature delTOutHis=1
+    "Delta between the temperature hysteresis high and low limit";
+  parameter Modelica.SIunits.SpecificEnergy delEntHis=1000
+    "Delta between the enthalpy hysteresis high and low limits, used if use_enthalpy = true"
+    annotation(Dialog(enable = use_enthalpy));
+  final parameter Modelica.SIunits.Temperature TOutHigLimCutHig = 0
     "Hysteresis high limit cutoff";
-  parameter Real uTemHigLimCutLow(
-    final unit="K", quantity = "ThermodynamicTemperature") = uTemHigLimCutHig - delTemHis
+  final parameter Real TOutHigLimCutLow = TOutHigLimCutHig - delTOutHis
     "Hysteresis low limit cutoff";
-  parameter Real uEntHigLimCutHig(final unit="J/kg", quantity="SpecificEnergy") = 0
+  final parameter Modelica.SIunits.SpecificEnergy hOutHigLimCutHig = 0
     "Hysteresis block high limit cutoff";
-  parameter Real uEntHigLimCutLow(final unit="J/kg", quantity="SpecificEnergy") = uEntHigLimCutHig - delEntHis
+  final parameter Real hOutHigLimCutLow = hOutHigLimCutHig - delEntHis
```

Comment:
These low limits need to be declared as ```Real``` as opposed to using ```Modelica.SIunits.*``` due to their negative values, which allow the sequence to be used both for differential and fixed cutoff temperature/enthalpy. Otherwise the inbuilt limits need to be breached and warnings get issued.